# 2. Cleaning
Obj: prepare dataset to EDA

In [2]:
import pandas as pd
df = pd.read_csv(r"C:\Users\Admin\personal-finance-analysis\data\personal_transactions.csv")

In [4]:
df["Date"] = pd.to_datetime(df["Date"])

In [10]:
df["Month"] = df["Date"].dt.to_period("M")
df["Year"] = df["Date"].dt.year
df["Weekday"] = df["Date"].dt.day_name()

In [14]:
df_clean = df[df["Category"] != "Credit Card Payment"].copy() 

163 rows where the payment category was “Credit Card Payment” were removed, as this category would have interfered with the analysis

In [18]:
expenses = df_clean[df_clean["Transaction Type"] == 'debit'].copy()
income = df_clean[df_clean["Transaction Type"] == 'credit'].copy()

In [19]:
print(f"Initial dataset lines: {len(df)}")
print(f"Cleaned dataset lines: {len(df_clean)}")
print(f"Number of expense transactions: {len(expenses)}")
print(f"Number of income transactions: {len(income)}")
print(f"Gaps in expenses: {expenses.isnull().sum().sum()}")

Initial dataset lines: 806
Cleaned dataset lines: 663
Number of expense transactions: 617
Number of income transactions: 46
Gaps in expenses: 0


Cleaning results:
- "Date" converted to datetime, extracted Month/Year/Weekday
- Deleted 163 lines 'Credit Card Payments'
- For analysis: 617 expense transactions, 46 income transactions
- No gaps found - the dataset is ready for EDA